projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -557,20 +548,23 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
      * @param rowCurrentValue Current value of the row as it is with applied previous transactions
      * @param rowNewValue New value of the row
      */
-    protected updateData(grid, rowID, rowValueInDataSource: any, rowCurrentValue: any, rowNewValue: { [x: string]: any }, validity?) {
-        
+    protected updateData(grid, rowID, rowValueInDataSource: any, rowCurrentValue: any, rowNewValue: { [x: string]: any }) {
         const transaction: Transaction = {
             id: rowID,
             type: TransactionType.UPDATE,
-            newValue: rowNewValue,
-            validity: validity
+            newValue: rowNewValue
         };
         if (grid.transactions.enabled) {
             grid.transactions.add(transaction, rowCurrentValue);
         } else {
             mergeObjects(rowValueInDataSource, rowNewValue);
         }
-        grid.transactions.addValidation(transaction, rowCurrentValue);
+
+        if (!this.grid.transactions.isPending) {
```

Comment:
Really not a fan of adding this as hidden implementation to the service (considering it's both public API and replaceable). There should be a good way to check if there's an active row edit (presuming that's what you're after) from the CRUD service, right?

Reply:
@damyanpetev 
I cannot think of a better place since we want to have this work with any api calls and all calls (api and ui) go through here. The only time we don't want to trigger an update is when this is a pending transaction that may be discarded in the future. 

The pending state change does not seem to be the same as the actual exiting of the edit mode, since exitCellEdit/exitRowEdit is called after the data is updated in the data/transactions. And we don't really track whether the current state is pending in the crud service (at least as far as I can tell).

Also, not sure if this is a viable scenario, but `startPending`/`endPending` are public apis, so in theory the user can use them to build some pending transactions himself. In that case any state we store in the crud service would not apply to those so we may trigger validation for something that is still pending, I guess. Not sure if viable, but still might be worth considering.